### ISSUE TITLE
temporal bug in clef changes

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#229]
I am trying to convert the following Humdrum data into MEI for verovio:

```
**kern
*clefG2
*M4/4
=1-
*^
4e          [1c
4g          .
4f          .
4e          .
=           =
4f          2c]
4b          .
4a          2d
4f          .
*v          *v
*clefF4
=
1C;
=
*-
```

It has a clef change at the end of the first measure so that the next measure is in bass clef.

The expected rendering in verovio:

![screen shot 2016-07-04 at 9 15 01 am](https://cloud.githubusercontent.com/assets/3487289/16566267/db348db2-41c7-11e6-832e-5251bed593b6.png)

But instead I get this rendering:

![screen shot 2016-07-04 at 9 12 35 am](https://cloud.githubusercontent.com/assets/3487289/16566272/e30d6e82-41c7-11e6-90e4-8aa68dd6f0c2.png)

Where the clef-change at the end of layer 1 is retroactively placing all of layer 2 in the bass clef (but superimposed on the same staff as treble clef music of layer 1).

This seems to be an error in verovio, but you can comment on how the MEI data is organized (with a clef change indicated at the end of layer 1).

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-04 09:18:00</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-b6a184c</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000105329191">
                        <staffGrp xml:id="m-000000202014990">
                            <staffDef xml:id="staffdef-000000094291023" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000121040439" n="1">
                            <staff xml:id="staff-000000159178472" n="1">
                                <!-- kern: =1-  *^  4e  4g  4f  4e  = -->
                                <layer xml:id="layer-000000192213281" n="1">
                                    <note xml:id="note-000000068975349" dur="4" oct="4" pname="e" accid.ges="n" />
                                    <note xml:id="note-000000057031438" dur="4" oct="4" pname="g" accid.ges="n" />
                                    <note xml:id="note-000000105426809" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <note xml:id="note-000000019636849" dur="4" oct="4" pname="e" accid.ges="n" />
                                </layer>
                                <!-- kern: [1c  = -->
                                <layer xml:id="layer-000000183043048" n="2">
                                    <note xml:id="note-000000134183408" dur="1" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <tie xml:id="tie-000000030617622" startid="note-000000134183408" endid="note-000000063594527" />
                        </measure>
                        <measure xml:id="measure-000000147968944">
                            <staff xml:id="staff-000000094252546" n="1">
                                <!-- kern: =  4f  4b  4a  4f  *v  *clefF4  = -->
                                <layer xml:id="layer-000000118614359" n="1">
                                    <note xml:id="note-000000042473967" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <note xml:id="note-000000036412551" dur="4" oct="4" pname="b" accid.ges="n" />
                                    <note xml:id="note-000000167667072" dur="4" oct="4" pname="a" accid.ges="n" />
                                    <note xml:id="note-000000052442552" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <clef xml:id="clef-000000074694500" shape="F" line="4" />
                                </layer>
                                <!-- kern: =  2c]  2d  *v -->
                                <layer xml:id="layer-000000186271509" n="2">
                                    <note xml:id="note-000000063594527" dur="2" oct="4" pname="c" accid.ges="n" />
                                    <note xml:id="note-000000053302897" dur="2" oct="4" pname="d" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-000000146372480">
                            <staff xml:id="staff-000000020017636" n="1">
                                <!-- kern: =  1C;  = -->
                                <layer xml:id="layer-000000140480896" n="1">
                                    <note xml:id="note-000000118909325" dur="1" fermata="above" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed in d8be65507c7a8cfebc2d997dedd036c8317e728e

**Comment by craigsapp** [comment]
Yeaaaay!  That makes piano music look much better.

![screen shot 2016-07-15 at 8 01 46 am](https://cloud.githubusercontent.com/assets/3487289/16878717/794b3396-4a62-11e6-9ee8-6aa906e95ebe.png)

I also like that the clef change does not disturb the spacing of the music, but ideally the clef change would be shifted a little to the right so that it is not as close to the note stem as it currently is.
